awpy/analytics/map_control.py

```diff
@@ -0,0 +1,413 @@
+                        for neighbor in neighbors
+                    ]
+                )
+
+    return map_control_values
+
+
+def _calc_frame_map_control_tile_values(
+    map_name: str,
+    ct_tiles: list[TileId],
+    t_tiles: list[TileId],
+    neighbor_info: TileNeighbors,
+) -> FrameMapControlValues:
+    """Calculate a frame's map control values for each side.
+
+    Values are allocated to tiles depending on how many tiles are between it
+    and the source tile (aka tile distance). The smaller the tile distance,
+    the close the tile's value is to 1. The tile distance can get as high as
+    the max_depth argument, which is 10 as default. If the max_depth is kept
+    as default 10, this means the BFS search will stop once a tile
+    distance of 10 is reached.
+
+    Args:
+        map_name (str): Map for other arguments
+        ct_tiles (list): List of CT-occupied tiles
+        t_tiles (list): List of T-occupied tiles
+        neighbor_info (TileNeighbors): Object with tile to neighbor mapping
+
+    Returns:
+        FrameMapControlValues object containing each team's map control values
+    """
+    return FrameMapControlValues(
+        t_values=_bfs(map_name, t_tiles, neighbor_info),
+        ct_values=_bfs(map_name, ct_tiles, neighbor_info),
+    )
+
+
+def graph_to_tile_neighbors(
+    neighbor_pairs: list[tuple[TileId, TileId]],
+) -> TileNeighbors:
+    """Convert list of neighboring tiles to TileNeighbors object.
+
+    Args:
+        neighbor_pairs (list): List of tuples (pairs of TileId)
+
+    Returns: TileNeighbors object with tile to neighbor mapping
+    """
+    tile_to_neighbors: TileNeighbors = defaultdict(set)
+
+    for tile_1, tile_2 in neighbor_pairs:
+        tile_to_neighbors[tile_1].add(tile_2)
+        tile_to_neighbors[tile_2].add(tile_1)
+
+    return tile_to_neighbors
+
+
+def calc_parsed_frame_map_control_values(
+    map_name: str,
+    current_player_data: FrameTeamMetadata,
+) -> FrameMapControlValues:
+    """Calculate tile map control values for each team given parsed frame.
+
+    Values are allocated to tiles depending on how many tiles are between it
+    and the source tile (aka tile distance). The smaller the tile distance,
+    the close the tile's value is to 1. The tile distance can get as high as
+    the max_depth argument, which is 10 as default. If the max_depth is kept
+    as default 10, this means the BFS search will stop once a tile
+    distance of 10 is reached.
+
+    Args:
+        map_name (str): Map used for find_closest_area and
+            relevant tile neighbor dictionary
+        current_player_data (FrameTeamMetadata): Object containing team metadata
+            (player positions, etc.). Expects extract_team_metadata output format
+
+    Returns: FrameMapControlValues object containing each team's map control values
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
```

Comment:
looks like this line was untested

Reply:
Added a simple sanity test for this line/function in next commit, thanks!